### PR TITLE
LPD-6701 Add clear mapping option

### PR DESCRIPTION
**Pull request by p2kmgcl**
This PR adds a new dropdown when some field has been mapped to a List visualization mode. The "change assignment" option works fine, but the "clear" one doesn't. It looks like the PATCH endpoint doesn't expect the fieldName to be null or empty, so it returns an error.

I tried to send a DELETE request to clear the mapping, but it seems like it is removing more elements than what it should...

### CODE REVIEW

**Comment by p2kmgcl** [comment]
@markocikos could you give me a hand with the clear process? I am not sure which endpoint should we be calling for that.

**Comment by markocikos** [comment]
> could you give me a hand with the clear process? I am not sure which endpoint should we be calling for that.

@p2kmgcl  There are two `DELETE` endpoints that would work, a better one is the one using ERC. You can check it in your [local API explorer](http://localhost:8080/o/api?endpoint=http://localhost:8080/o/data-set-manager/list-sections/openapi.json). See screen below.

In theory, we could also set field value to an empty string, if we wanted to distinguish between field name that was never set and field that was cleared, but I don't think there is a reason for that. It would only complicate overall implementation.  

image

**Comment by p2kmgcl** [comment]
Thanks @markocikos! Now it works!

File: modules/apps/frontend-data-set/frontend-data-set-views-web/src/main/resources/META-INF/resources/js/fds_view/visualization_modes/list/List.tsx [hunk 1]
```diff
@@ -256,6 +296,48 @@ function ListSection({
...
+		<ClayButtonWithIcon
+			aria-label={Liferay.Language.get('assign-field')}
+			displayType="secondary"
+			onClick={openSelectFieldModal}
+			symbol="plus"
+			title={Liferay.Language.get('assign-field')}
+		/>
+	);
+
+	if (field) {
```
Comment (markocikos) [inline modules/apps/frontend-data-set/frontend-data-set-views-web/src/main/resources/META-INF/resources/js/fds_view/visualization_modes/list/List.tsx:309]: This seems functional, but it's not very React-y. A cleaner solution would be a new "private" component, something like `<UpdateButton>`, that would internally `return field ? <first version> : <second version>`

**Review by markocikos — APPROVED** [review]
@p2kmgcl Looks good, non-blocking improvement suggestion inline.

File: modules/apps/frontend-data-set/frontend-data-set-views-web/src/main/resources/META-INF/resources/js/fds_view/visualization_modes/list/List.tsx (hunk 1 above)
Comment (p2kmgcl) [reply modules/apps/frontend-data-set/frontend-data-set-views-web/src/main/resources/META-INF/resources/js/fds_view/visualization_modes/list/List.tsx]: Done!

**Comment by p2kmgcl** [comment]
ci:forward

**Comment by p2kmgcl** [comment]
ci:forward

**Comment by markocikos** [comment]
ci:test:stable

**Comment by markocikos** [comment]
ci:forward:force

**Pull request closed without merging**